pkg/engine/jmespath/functions.go

```diff
@@ -965,3 +980,26 @@ func jpRandom(arguments []interface{}) (interface{}, error) {
 	}
 	return ans, nil
 }
+
+func jpX509Decode(arguments []interface{}) (interface{}, error) {
+	p, _ := pem.Decode([]byte(arguments[0].(string)))
+	if p == nil {
+		return "", errors.New("invalid certificate")
+	}
+
+	var v interface{}
+	cert, err := x509.ParseCertificate(p.Bytes)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+	v = struct{ *x509.Certificate }{cert}
+
+	buf := new(bytes.Buffer)
+	enc := json.NewEncoder(buf)
+	err = enc.Encode(v)
+	if err != nil {
+		return "", errors.WithStack(err)
+	}
+
+	return strings.TrimSuffix(buf.String(), "\n"), nil
```

Comment:
I believe we want the output to be query-able. We should return a map[string]interface here instead of a string.

Reply:
✅

Reply:
@samj1912 what would be the problem with a JSON string here? The general thought is that this filter would just generate the JSON to be piped to other filters for further extraction/processing.

Reply:
@chipzoller the issue is that the user will have to almost always pair x509_decode with parse_json in order to do anything further with it.

Given that we are decoding the certificate, it's better to give back something query-able rather than a string.

Reply:
I see, so you'd do something like x509_decode.somefield rather than inserting parse_json in between, making it a little less cumbersome?

Reply:
yup, exactly!